Profiles/cio-SurveillanceDecision.xml

```diff
@@ -0,0 +1,267 @@
+      <path value="Flag.code.extension" />
+      <sliceName value="safeWithinUnsafeGroup" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-SurveillanceDecision.SafeWithinUnsafeGroup" />
+      </type>
+    </element>
+    <element id="Flag.code.text">
+      <path value="Flag.code.text" />
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231215" />
+        <map value="cio-dataelement-20-760" />
+        <comment value="SubstanceForSurveillance (implicit)" />
+      </mapping>
+    </element>
+    <element id="Flag.subject">
+      <path value="Flag.subject" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Patient" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-Patient" />
+      </type>
+    </element>
+    <element id="Flag.period">
+      <path value="Flag.period" />
+      <min value="1" />
+    </element>
+    <element id="Flag.period.start">
+      <path value="Flag.period.start" />
+      <short value="DecisionEffectiveDateTime" />
+      <definition value="Moment (date and time) when the decision (of type 'start surveillance') should take effect." />
+      <comment value="When the value of DecisionType is _385652002_ the DecisionEffectiveDateTime should be mapped to `.period.start`. Whenever the DecisionType is _410546004_, the DecisionEffectiveDateTime should be mapped to `.period.end`, and the `.period.start` should not change (i.e. it should be the same as in the Flag resource that is referenced via `.extension:relationSurveillanceDecision`)." />
+      <alias value="BesluitIngangsDatumTijd" />
+      <min value="1" />
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231215" />
+        <map value="cio-dataelement-20-757" />
+        <comment value="DecisionEffectiveDateTime" />
+      </mapping>
+    </element>
+    <element id="Flag.period.end">
+      <path value="Flag.period.end" />
+      <short value="DecisionEffectiveDateTime" />
+      <definition value="Moment (date and time) when the decision (of type 'stop surveillance') should take effect." />
+      <comment value="When the value of DecisionType is _385652002_ the DecisionEffectiveDateTime should be mapped to `.period.start`. Whenever the DecisionType is _410546004_, the DecisionEffectiveDateTime should be mapped to `.period.end`, and the `.period.start` should not change (i.e. it should be the same as in the Flag resource that is referenced via `.extension:relationSurveillanceDecision`)." />
+      <alias value="BesluitIngangsDatumTijd" />
+      <condition value="cio-SurveillanceDecision-1" />
+      <mapping>
+        <identity value="cio-dataset-200-beta2-20231215" />
+        <map value="cio-dataelement-20-757" />
+        <comment value="DecisionEffectiveDateTime" />
+      </mapping>
+    </element>
```

Comment:
Dit blijkt verder niet echt uit TO, maar ik haal hieruit dat een eenmaal geregistreerd BewakingsBesluit niet meer wijzigt en dat je (a la MP) een nieuwe instance (en dus een nieuwe Flag) krijgt bij een besluit te stoppen.

Gaat dat niet fundamenteel in tegen hoe een Flag op FHIR-niveau zou moeten werken? Als iemand zoekt op actieve Flags, dan verwacht je ... actieve Flags, niet Flags die status actief hebben maar die mogelijk in een andere Flag ongeldig worden verklaard - in een andere Flag waarmee een relatie wordt gelegd in een extensie nota bene.

Het verschil met bijvoorbeeld MP bouwstenen zit hem hier denk ik in de aard van de resource en dat is misschien nogal subjectief hoor.

Reply:
Dit neem ik mee in de Discussie binnen het TO.